onnx/backend/test/case/node/averagepool.py

```diff
@@ -0,0 +1,226 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+
+import onnx
+from ..base import Base
+from . import expect
+from .pool_op_common import get_output_shape, get_pad_shape, pool
+
+
+class AveragePool(Base):
+
+  @staticmethod
+  def export():
+
+    """
+    output_shape: [1, 1, 2, 2]
+    """
+    node = onnx.helper.make_node(
+      'AveragePool',
+      inputs=['x'],
+      outputs=['y'],
+      kernel_shape=[2, 2],
+      strides=[2, 2]
+    )
+    x = np.array([[[
+      [1, 2, 3, 4, 5],
+      [6, 7, 8, 9, 10],
+      [11, 12, 13, 14, 15],
+      [16, 17, 18, 19, 20],
+      [21, 22, 23, 24, 25],
+    ]]]).astype(np.float32)
+    y = np.array([[[[4, 6],
+                    [14, 16]]]]).astype(np.float32)
+
+    expect(node, inputs=[x], outputs=[y], name='test_averagepool_2d_precomputed_strides')
+    
+    """
+    output_shape: [1, 1, 3, 3]
+    pad_shape: [2, 2] -> [1, 1, 1, 1] by axis
+    """
+    node = onnx.helper.make_node(
+      'AveragePool',
+      inputs=['x'],
+      outputs=['y'],
+      kernel_shape=[3, 3],
+      strides=[2, 2],
+      auto_pad='SAME_UPPER'
```

Comment:
Because auto_pad is deprecated, please provide (additional) examples with explicit padding.